owncloudApp/src/main/java/com/owncloud/android/presentation/sharing/shares/PublicShareDialogFragment.kt

```diff
@@ -321,6 +343,70 @@ class PublicShareDialogFragment : DialogFragment() {
         })
     }
 
+    private fun updateRequirements(password: String, textView: TextView) {
+        val hasMaxCharacters = password.length <= 72
+        val hasMinCharacters = password.length >= 8
+        val hasUpperCase = password.any { it.isUpperCase() }
+        val hasLowerCase = password.any { it.isLowerCase() }
+        val hasSpecialCharacter = password.any { SPECIALS_CHARACTERS.contains(it) }
+        val hasDigit = password.any { it.isDigit() }
+
+        val passwordChecks = arrayOf(
+            Triple(hasMinCharacters, R.string.password_policy_min_characters, capabilities?.passwordPolicyMinCharacters),
+            Triple(hasUpperCase, R.string.password_policy_uppercase_characters, capabilities?.passwordPolicyMinUppercaseCharacters),
+            Triple(hasLowerCase, R.string.password_policy_lowercase_characters, capabilities?.passwordPolicyMinLowercaseCharacters),
+            Triple(hasDigit, R.string.password_policy_min_digit, capabilities?.passwordPolicyMinDigits),
+            Triple(hasSpecialCharacter, R.string.password_policy_min_special_character, capabilities?.passwordPolicyMinSpecialCharacters),
+            Triple(hasMaxCharacters, R.string.password_policy_max_characters, capabilities?.passwordPolicyMaxCharacters)
+        )
+
+        val requirementsString = buildSpannedString {
+
+            appendLine(getString(R.string.password_policy_intro))
+
+            for ((requirementIsChecked, resourceId, capabilityValue) in passwordChecks) {
+                capabilityValue?.let {
+                    val requirement = getString(
+                        resourceId,
+                        capabilityValue.toString(),
+                        if (requirementIsChecked == hasSpecialCharacter) SPECIALS_CHARACTERS else ""
+                    )
+                    appendRequirement(requirementIsChecked, requirement)
+                }
+            }
+        }
+
+        textView.text = requirementsString
+
+        val allConditionsCheck = hasMinCharacters && hasUpperCase && hasLowerCase && hasDigit && hasSpecialCharacter && hasMaxCharacters
+        if (binding.shareViaLinkPasswordPolicy.isVisible) {
+            binding.saveButton.isEnabled = allConditionsCheck
+        }
+    }
+
+    private fun SpannableStringBuilder.appendRequirement(requirementIsChecked: Boolean, requirement: String) {
```

Comment:
I don't really like this. I think it's much more confusing this than managing it in a visual mode, with one TextView for each requirement, and handling its visibility depending on whether the restriction is present or not

Reply:
Done